### DISCUSSION TITLE
Streaming DJ'ing (Serato)

### DISCUSSION THREAD

**Discussion by rewindthedj** [ExistentialAudio/BlackHole#106] [General]
I saw someone commented re: DJing with Traktor, but I`m trying to use BlackHole to reroute my Serato audio to OBS. 
I used the trial of LoopBack, and although a lot of people have latency issues, it did exactly what I needed it to do in regards to the audio quality in my stream.
Any other solutions that I have tried have resulted in very distorted audio, or no audio at all.

In summary, LoopBack has an option to stream the Serato window's audio to OBS and I`m wondering if I can get the same capability from BlackHole?

**Comment by krono-i2** [comment]
If you are using Serato I think you have an audio interface (or integrated controller) to unlock software features. Which one do you have?
Feaseability and optimal configuration could vary depending on this.

**Comment by rewindthedj** [comment]
I have a DDJ-SR1

**Comment by krono-i2** [comment]
Unfortunately with Serato DJ is not possible to set your own interface, it’s locked on the primary Serato hardware connected.
If your Mac has line audio input (my MacBook Pro mid 2009 has it) you can implement an analog routing as in https://support.serato.com/hc/en-us/articles/202304744-Recording-your-mix-with-Audacity
Alternatively, there is a bug in Serato that permits to use custom interfaces naming an aggregate device with the same name of Serato audio interfaces (I tested it, and it works with rane sl2). The main problem of this solution is that Blackhole, at this moment, does not implement channels routing.

[EDIT]
If you have an rca stereo cable try to connect the rca main (2) out to aux in, and see if OBS receive audio setting the pioneer interface as input device.
If aux input is in-driver associated with the input channels 1-2 it could work.
Let me know.

**Comment by anthonyhashemi** [comment]
+1

**Comment by djquantify** [comment]
Thanks for the driver @krono86 

Does blackhole allow custom mapping of channels?  My mixing console outputs on channels 9&10.  Using a different utility, I was able to capture the audio from my DJ mixer directly via USB by mapping the 9&10 channel to the 1&2 output. But I'm still trying to figure out a way to do it with BH

**Comment by krono-i2** [comment]
Not at the moment.
It seems that channel routing is “under considering” for future improvements.

**Comment by djquantify** [comment]
Thanks for the update.
